src/librbd/ExclusiveLock.h

```diff
@@ -94,13 +95,8 @@ class ExclusiveLock {
     STATE_LOCKED,
     STATE_INITIALIZING,
     STATE_ACQUIRING,
-    STATE_POST_ACQUIRING,
```

Comment:
I'm still not convinced that the _PRE/_POST states are required. Regarding the release of the exclusive-lock, in the current implementation, the managed-lock is only released after the object_map and journal are closed.
Regarding the acquire of the exclusive-lock, the managed-lock is acquired before the object-map and journal are opened. The only problem I found was with the RefreshRequest that would see that the managed-lock was already acquired (although the exclusive-lock was still in m_state == STATE_ACQUIRING) , and would try to open the object-map and journal, which conflicted with the open of object-map and journal in the AcquireRequest class.